### DISCUSSION TITLE
Coverting carriage return in individual segments

### DISCUSSION THREAD

**Discussion by rutledj** [nextgenhealthcare/connect#6090] [Q&A]
What is the proper way to replace \X0d0a\  with \.br\ in javascript on a per segment basis?

If I put: this in the preprocessor, it works:
```
message = message.replace(/\\X0d0a\\/g, '\\.br\\');
```

But if I try this in a transformer it does not.

```
for each(seg in msg.children())
{
	seg = seg.toString().replace(/\\X0d0a\\/g, '\\.br\\');
	
}
```

**Comment by rjkroll** [comment]
If it works in the pre-processer, why not just use it there?  Also, I believe the issue might be related to what type you are trying to do a replace on in the transformer step, which usually means either you are trying to do something on a Java string type that only works on Javascript string types, or vice versa.  Try removing the .toString() in the transformer part to see if that works.

**Comment by rutledj** [comment]
Unfortunately, it causes problems with NTE segments in non-discrete micro results. There are no carriage returns after the individual NTE segments, so they all run together in the receiving system.

**Comment by jonbartels** [comment]
Would a regex like the attachment handler be useful?

`(?:OBX\|(?:[^|]*?\|){4}(?:[^|^]*?\^){4})([^|^]*)[|^]` from [this article](https://forums.mirthproject.io/articles/175046-using-attachments) might be a good start. Since it has matching groups I think it is more complicated than you need.

[I tried to write one](https://regex101.com/r/GXhoBK/1) but it does not work quite right. Can you fiddle with this example and share a revision?
